core/src/main/scala/com/salesforce/op/stages/impl/classification/BinaryClassificationModelSelector.scala

```diff
@@ -106,90 +185,50 @@ case object BinaryClassificationModelSelector {
-      // Logistic regression defaults
-      .setLogisticRegressionElasticNetParam(ElasticNet)
-      .setLogisticRegressionFitIntercept(FitIntercept)
-      .setLogisticRegressionMaxIter(MaxIterLin)
-      .setLogisticRegressionRegParam(Regularization: _*)
-      .setLogisticRegressionStandardization(Standardized)
-      .setLogisticRegressionTol(Tol)
-      // NB defaults
-      .setNaiveBayesModelType(NbModel)
-      .setNaiveBayesSmoothing(NbSmoothing)
-      // DT defaults
-      .setDecisionTreeImpurity(ImpurityClass)
-      .setDecisionTreeMaxBins(MaxBin)
-      .setDecisionTreeMaxDepth(MaxDepth: _*)
-      .setDecisionTreeMinInfoGain(MinInfoGain: _*)
-      .setDecisionTreeMinInstancesPerNode(MinInstancesPerNode: _*)
+    )
   }
 }
 
 /**
- * Binary Classification Model Selector
- *
- * @param validator Cross Validation or Train Validation Split
- * @param splitter  instance that will balance and/or split the data
- * @param evaluators List of evaluators applied on training + holdout data for evaluation.
- * @param uid
+ * Enumeration of possible classification models in Model Selector
  */
-private[op] class BinaryClassificationModelSelector
-(
-  override val validator: OpValidator[ProbClassifierModel, ProbClassifier],
-  override val splitter: Option[Splitter],
-  override val evaluators: Seq[OpBinaryClassificationEvaluatorBase[_ <: EvaluationMetrics]],
-  override val uid: String = UID[BinaryClassificationModelSelector]
-) extends ClassificationModelSelector(validator, splitter, evaluators, uid) with StageOperationName {
-
-  override private[classification] val stage1uid: String = UID[Stage1BinaryClassificationModelSelector]
-
-  lazy val stage1 = new Stage1BinaryClassificationModelSelector(validator = validator, splitter = splitter,
-    evaluators = evaluators, uid = stage1uid, stage2uid = stage2uid, stage3uid = stage3uid)
-}
+sealed trait BinaryClassificationModelsToTry extends ModelsToTry
 
-/**
- * Stage 1 of BinaryClassificationModelSelector
- *
- * @param validator Cross Validation or Train Validation Split
- * @param splitter  instance that will balance and split the data
- * @param evaluators List of evaluators applied on training + holdout data for evaluation.
- * @param uid
- */
-private[op] class Stage1BinaryClassificationModelSelector
-(
-  validator: OpValidator[ProbClassifierModel, ProbClassifier],
-  splitter: Option[Splitter],
-  evaluators: Seq[OpBinaryClassificationEvaluatorBase[_ <: EvaluationMetrics]],
-  uid: String = UID[Stage1BinaryClassificationModelSelector],
-  stage2uid: String = UID[SwTernaryTransformer[_, _, _, _, _]],
-  stage3uid: String = UID[SwQuaternaryTransformer[_, _, _, _, _, _]]
-) extends Stage1ClassificationModelSelector(validator, splitter, evaluators, uid, stage2uid, stage3uid)
+object BinaryClassificationModelsToTry extends Enum[BinaryClassificationModelsToTry] {
+  val values = findValues
+  case object OpLogisticRegression extends BinaryClassificationModelsToTry
```

Comment:
Should we associate each BinaryClassificationModelsToTry with the corresponding estimator and default params?
```
case object OpLogisticRegression extends BinaryClassificationModelsToTry {
  val estimator = new OpLogisticRegression()
  val params = new ParamGridBuilder()
      .addGrid(estimator.fitIntercept, DefaultSelectorParams.FitIntercept)
      .addGrid(estimator.elasticNetParam, DefaultSelectorParams.ElasticNet)
      .addGrid(estimator.maxIter, DefaultSelectorParams.MaxIterLin)
      .addGrid(estimator.regParam, DefaultSelectorParams.Regularization)
      .addGrid(estimator.standardization, DefaultSelectorParams.Standardized)
      .addGrid(estimator.tol, DefaultSelectorParams.Tol)
      .build()
}
```
Also, we might let users define custom BinaryClassificationModelsToTry?

Reply:
I can make the base class public for BinaryClassificationModelsToTry. And I could add a sub value to have the class they are associated with rather than relying on the name...